hangman.py

```diff
@@ -0,0 +1,370 @@
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    found = True
+    for i in secret_word:
+        if i not in letters_guessed:
+            found = False
+    return found
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    result = ""
+    for i in secret_word:
+        if i not in letters_guessed:
+            result += "_ "
+        elif i in letters_guessed:
+            result += i
+    return result
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    letters = list(string.ascii_lowercase)
+    result = ""
+    for i in letters_guessed:
+        letters.remove(i)
+    for i in letters:
+        result += i
+    return result
+
+
+def check_warnings(warnings, user_guess, repeated_guesses, show_word):
+    '''
+
+    :param warnings: warnings that are given to user
+    :param user_guess: user's input
+    :param repeated_guesses: letters that have been repeated
+    :param show_word: output that is visible to user
+    :return: warnings that are left
+    '''
+    warnings -= 1
+    if not user_guess.isalpha():
+        print('Oops! That is not a valid letter. You have ' + str(warnings) + ' warnings left: ' + show_word)
+    elif user_guess in repeated_guesses:
+        print('Oops! You\'ve already guessed that letter. You have ' + str(warnings) + ' warnings left. ' + show_word)
+    print('-----------------')
+    return warnings
+
+
+def check_guesses(guesses, user_guess, repeated_guesses, show_word):
+    '''
+
+    :param guesses: guesses that are given to user
+    :param user_guess: user's input
+    :param repeated_guesses: letters that have been repeated
+    :param show_word: output that is visible to user
+    :return: warnings that are left
+    '''
+    guesses -= 1
+    if not user_guess.isalpha():
+        print("Oops! That is not a valid letter. You have no warnings left so you lose one guess. " + show_word)
+    elif user_guess in repeated_guesses:
+        print(
+            'Oops! You\'ve already guessed that letter. You have no warnings left so you lose one guess: ' + show_word)
+    else:
+        print('Oops! That letter is not in my word: ' + show_word)
+    print('--------')
+    return guesses
+
+
+def hangman(secret_word):
```

Comment:
I also want to highlight that current function is quite long - it worth to make it shorter by extracting some functions.